src/lib/parsers/sol-json.c

```diff
@@ -512,23 +512,15 @@ sol_json_escape_string(const char *str, char *buf, size_t len)
     return out;
 }
 
-SOL_API int
-sol_json_double_to_str(const double value, char *buf, size_t len)
+static void
+double_to_str(char *buf, size_t len)
```

Comment:
this is an internal function, not exported and used internally by sol_json_double_str. If it's still a problem i don't see any problem in rename it.

For the string size, i did not know the necessary size and as the function is using %f in the snprintf, 71 bytes were not enough to convert DBL_MAX.